packages/components/src/font-size-picker/utils.ts

```diff
@@ -142,13 +145,26 @@ export function getToggleGroupOptions(
 
 export function getSelectedOption(
 	fontSizes: FontSize[],
-	value: FontSizePickerProps[ 'value' ]
+	value: FontSizePickerProps[ 'value' ],
+	useSelectControl: boolean,
+	disableCustomFontSizes: boolean = false
 ): FontSizeOption {
 	if ( ! value ) {
 		return DEFAULT_FONT_SIZE_OPTION;
 	}
-	return (
-		fontSizes.find( ( font ) => font.size === value ) ||
-		CUSTOM_FONT_SIZE_OPTION
+
+	const fontSizeOptions = getFontSizeOptions(
+		useSelectControl,
+		fontSizes,
+		disableCustomFontSizes
 	);
+
+	const selectedOption = fontSizeOptions
+		? // @ts-ignore @TODO fix types for fontSizeOptions. Array.find() works on the same type only.
```

Comment:
It looks like this may be related to a Typescript issue with union types: https://github.com/microsoft/TypeScript/issues/44373

I don't think there's currently a fix but thought it might be useful to just drop the link here for future reference.

Reply:
Thanks for finding that. 

Yeah, it was doing my head in. I tried all sorts of unions before realizing that the impenetrable 100-line-long Typescript error was really wah wahing for the same object 🙃 

I'll add a note as you say. Cheers!